test/distributed/_composable/test_checkpoint.py

```diff
@@ -184,17 +174,59 @@ def test_multi_args(self, use_reentrant: bool):
             MultiInputModel(device),
         )
         net2 = deepcopy(net1)
-        checkpoint(net2[0], use_reentrant=use_reentrant)
-        checkpoint(net2[2], use_reentrant=use_reentrant)
+        checkpoint(net2[0])
+        checkpoint(net2[2])
         x1 = torch.randn(20, 100, requires_grad=True)
         x2 = x1.clone()
         net1(x1).sum().backward()
         net2(x2).sum().backward()
         for p1, p2 in zip(net1.parameters(), net2.parameters()):
             self.assertEqual(p1.grad, p2.grad)
 
+    def test_raise_in_forward(self):
+
+        class MyModel(torch.nn.Module):
+            def __init__(self, raise_in_recomp):
+                super().__init__()
+                self.fwd_count = 0
+                self.raise_in_recomp = raise_in_recomp
+                self.a = torch.nn.Linear(2, 2)
+
+            def forward(self, x):
+                if self.raise_in_recomp and self.fwd_count == 1:
+                    raise RuntimeError("foo")
+                else:
+                    if not self.raise_in_recomp:
+                        # raise in the first forward
+                        raise RuntimeError("foo")
+                    self.fwd_count += 1
+                    return self.a(x)
+
+        m = MyModel(raise_in_recomp=True)
+        checkpoint(m)
+        inp = torch.randn(1, 2)
+        # Should raise in forward recomputation
+        out = m(inp).sum()
+
+        with self.assertRaises(RuntimeError):
```

Comment:
```suggestion
        with self.assertRaisesRegex(RuntimeError, "foo"):
```
(or factor the runtime error message into a local variable and use that)